### PR TITLE
Docker build for internal barista speedup

### PR DESCRIPTION
**Pull request by lukasholzer**
### <strong>Pull Request</strong>

This pull request includes the Internal release on our Jenkins that is building the design system with all internal resources and deploying it to our internal registry. 

After everything is done it will tell Kubernetes to use the built image as the current image.

The `.deployment/barista/k8s-barista-builder.Dockerfile` Dockerfile is the Jenkins agent that is used by the Jenkins file as `k8s-barista-builder` to run the build inside.

This docker image is currently built by hand with:

```bash
docker build --no-cache -t ${DOCKER_REGISTRY}/designops/jenkins-barista-builder:latest . -f ./.deployment/barista/k8s-barista-builder.Dockerfile --build-arg DOCKER_REGISTRY=${DOCKER_REGISTRY}
```

This will be tackled in a follow-up issue to be handled by Jenkins as well.

image

### CODE REVIEW

**Comment by lukasholzer** [comment]
Jenkins run on the CI

image

File: .deployment/barista/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+                configFile(fileId: "barista-env", variable: "ENV")
+              ]) {
+                sh '''
+                  mv "$ENV" .env
+
+                  # Set public build to the inverse of INTERNAL_VERSION
+                  if $INTERNAL_VERSION ; then
+                    PUBLIC_BUILD=false
+                  else
+                    PUBLIC_BUILD=truedocker
```
Comment (ffriedl89) [inline .deployment/barista/Jenkinsfile:86]: is this supposed to be `truedocker` ? I guess not? :D

File: .deployment/barista/barista-internal.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+FROM nginx:alpine
+
+COPY .deployment/barista/nginx/nginx.conf /etc/nginx/nginx.conf
+COPY .deployment/barista/nginx/sites-available /etc/nginx/sites-available
+
+RUN ln -s /etc/nginx/sites-available /etc/nginx/sites-enabled
```
Comment (ffriedl89) [inline .deployment/barista/barista-internal.Dockerfile:7]: I have no idea about nginx - but is this supposed to be different - available - enabled?

Comment (lukasholzer) [reply .deployment/barista/barista-internal.Dockerfile:7]: jep in Nginx configurations you store your config for the available sites and then link the one that are currently enabled.

File: .deployment/barista/Jenkinsfile (hunk 1 above)
Comment (lukasholzer) [reply .deployment/barista/Jenkinsfile:86]: haha nope :D typo :D

**Comment by tomheller** [comment]
Great work. Build on circleCI seems to fail, maybe you can have a look at that.

**Pull request merged**